src/ng/compile.js

```diff
@@ -814,6 +814,52 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
 
   /**
    * @ngdoc method
+   * @name $compileProvider#component
+   * @kind function
+   *
+   * @description
+   * Register a new component with the compiler.
+   *
+   * A component is a shorthand for a directive with a controller, template, and a
+   * new scope.
+   *
+   * @returns {ng.$compileProvider} Self for chaining.
+   */
+  this.component = registerComponent;
+  function registerComponent(name, bindings, controller) {
+    if (isString(name)) {
+      assertValidDirectiveName(name);
```

Comment:
I'd rather see the error thrown in the registration method I call, rather than the underlying implementation.